pkgs/development/compilers/urn/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchFromGitLab, buildEnv, makeWrapper, lua, luajit, readline
+, useLuaJit ? false
+, extraLibraries ? []
+}:
+
+let
+  urnVersion = "0.7.1";
+  # Build a sort of "union package" with all the native dependencies we
+  # have: Lua (or LuaJIT), readline, etc. Then, we can depend on this
+  # and refer to ${urn-rt} instead of ${lua}, ${readline}, etc.
+  urn-rt = buildEnv {
+    name = "urn-rt-${urnVersion}";
+    ignoreCollisions = true;
+    paths = if useLuaJit then
+              [ luajit readline ]
+            else
+              [ lua ];
+  };
+
+  inherit (stdenv.lib) optionalString concatMapStringsSep;
+in
+
+stdenv.mkDerivation rec {
+  name = "urn-${optionalString (extraLibraries != []) "with-libraries-"}${version}";
+  version = urnVersion;
+
+  src = fetchFromGitLab {
+    owner = "urn";
+    repo = "urn";
+    rev = "381208dcaa651569d1de689384364775fa039c1c";
```

Comment:
Can you use `rev = "v${version}"` instead? This makes it easier to update the package automatically.